src/plugins/discover/public/application/main/utils/resolve_data_view.ts

```diff
@@ -70,7 +70,7 @@ export async function loadDataView({
   let fetchedDataView: DataView | null = null;
   // try to fetch adhoc data view first
   try {
-    fetchedDataView = fetchId ? await dataViews.get(fetchId, false, true) : null;
+    fetchedDataView = fetchId ? await dataViews.get(fetchId) : null;
```

Comment:
Additionally to this change, could you also take care of this issue in this PR? 

#169360

in `discover_main_route`, after the saved search is being loaded

https://github.com/elastic/kibana/blob/2f576621aa5673ab6ef32da9bd713de1dba348c1/src/plugins/discover/public/application/main/discover_main_route.tsx#L148-L152

its data view fields should be refreshed. Then there would be a loading spinner displayed for this case, which is better UX.

Reply:
We spoke about this and decided to put further changes into a separate PR.